packages/react-native-web/src/exports/Image/index.js

```diff
@@ -166,7 +166,9 @@ class Image extends Component<*, State> {
       ImageUriCache.remove(prevUri);
       const isPreviouslyLoaded = ImageUriCache.has(uri);
       isPreviouslyLoaded && ImageUriCache.add(uri);
-      this._updateImageState(getImageState(uri, isPreviouslyLoaded));
+      this._updateImageState(getImageState(uri, isPreviouslyLoaded), !!this.props.defaultSource);
```

Comment:
Please can we avoid coercion. And since this check is used 3 times we can lift up to a var: `const hasDefaultSource = this.props.defaultSource != null`